satellite/gracefulexit/endpoint.go

```diff
@@ -443,6 +442,23 @@ func (endpoint *Endpoint) doProcess(stream processStream) (err error) {
 		case *pb.StorageNodeMessage_Succeeded:
 			err = endpoint.handleSucceeded(ctx, stream, pending, nodeID, m)
 			if err != nil {
+				if metainfo.NodeAlreadyExitsError.Has(err) {
+					// this will get re-tried
+					endpoint.log.Warn("node already exists in pointer.", zap.Error(err))
+
+					if transfer, ok := pending.get(m.Succeeded.OriginalPieceId); ok {
+						endpoint.log.Debug("retrying peice b/c the replacement node already has a piece for the segment.", zap.Stringer("nodeID", nodeID), zap.ByteString("path", transfer.path), zap.Int32("pieceNum", transfer.pieceNum))
+						incomplete, err := endpoint.db.GetTransferQueueItem(ctx, nodeID, transfer.path, transfer.pieceNum)
+						if err != nil {
+							return rpcstatus.Error(rpcstatus.Internal, Error.Wrap(err).Error())
+						}
+						err = endpoint.processIncomplete(ctx, stream, pending, incomplete)
```

Comment:
Are we calling `processIncomplete` again for other similar cases where there is some internal error? Why not just wait for the incomplete transfer cycle pick it up in the next iteration?

Reply:
You are right. Changed to remove from the pending queue, so it gets reprocessed.